Add gitlabci.yml file

cmft opened the pull request:
Add gitlabci.yml file for CI/packaging purposes.

cmft committed:
Add .gitlab-ci.yml file

cmft committed:
Remove print

cpascual commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,144 @@
+  DOCKER_DRIVER: overlay2
+  ARTIFACTS_DIR: "$CI_PROJECT_DIR/dist"
+  # Set up variables for pushing the generated tags
+  DEPLOY_GIT_REPO: "git.cells.es/$CI_PROJECT_PATH.git"
```
I would eliminate ALBA-specific hardcodings.
Alternatively, I would callthis file `.gitlab-ci-alba` to avoid it automatically triggering pipelines that won't work if someone else pushes it to their own gitlab.

cpascual replied:
The easiest is to simply comment that the `DEPLOY_GIT_REPO` variable should be set in the gitlab project settings

cpascual commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,144 @@
+  # ALBA HACK:
+  GIT_SSL_NO_VERIFY: 'true'
+  DEBFULLNAME: "CI Boot"
+  DEBEMAIL: "controls@cells.es"
```
DEBEMAIL should also not hardcode this alba-specific value

cpascual commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,144 @@
+
+
+upload_to_staging:
+  image: dockerrg.cells.es/debpack-ci:latest
```
This image is only in the alba docker registry. If it cannot be substituted by a publicly available one, we should use a gitlab-project defined variable. 

Same for promote_to_production:

cmft replied:
It can be managed via variables expressions but unfortunate our gitlab version does not support
```
  only:
    variables:
      - $REPO_URL == 'git.cells.es'
```

cpascual replied:
Right... but maybe we can use :

```
only:
    - branches@ctpkg/VacuumController
```

cmft committed:
Improve .gitlab-ci.yml file

* Avoid hardcode ALBA configuration
* Ignore new in build command

cmft committed:
Avoid to hardcode debemail

cmft committed:
Hide some specific ALBA jobs

upload_to_staging and promote_to_production job are ALBA specific avoid their execution out of ALBA
